### PR TITLE
Add docs for using Grpc.Tools on unsupport platforms

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/grpc/grpc-dotnet/issues/2048

FYI @tonydnewell

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/grpc/troubleshoot.md](https://github.com/dotnet/AspNetCore.Docs/blob/bceca0cb2b415e434a6ebba83abcbcd45f8963b1/aspnetcore/grpc/troubleshoot.md) | [Troubleshoot gRPC on .NET](https://review.learn.microsoft.com/en-us/aspnet/core/grpc/troubleshoot?branch=pr-en-us-29314) |

### CODE REVIEW

File: aspnetcore/grpc/troubleshoot.md [hunk 1]
```diff
@@ -284,6 +284,50 @@ var reply = await client.SayHelloAsync(new HelloRequest { Name = ".NET" });
...
+# e.g. for Alpine Linux the grpc-plugins package can be used
+#  See https://pkgs.alpinelinux.org/package/edge/community/x86_64/grpc-plugins
+apk add grpc-plugins  # Alpine Linux specific package installer
+
+# Set environment variables for the built/installed protoc
+# and grpc_csharp_plugin binaries
+export PROTOBUF_PROTOC=/usr/bin/protoc
+export GRPC_PROTOC_PLUGIN=/usr/bin/grpc_csharp_plugin
+
+# When the dotnet build runs the Grpc.Tools NuGet package will
```
Comment (eerhardt) [inline aspnetcore/grpc/troubleshoot.md:324]: (nit) I'm not sure `the` makes sense here.

```suggestion
# When dotnet build runs, the Grpc.Tools NuGet package will
```

File: aspnetcore/grpc/troubleshoot.md [hunk 2]
```diff
@@ -284,6 +284,50 @@ var reply = await client.SayHelloAsync(new HelloRequest { Name = ".NET" });
 
 Alternatively, a client factory can be configured with `Http3Handler` by using <xref:Microsoft.Extensions.DependencyInjection.HttpClientBuilderExtensions.AddHttpMessageHandler%2A>.
 
+## Building gRPC on Alpine Linux
+
+The `Grpc.Tools` package [generates .NET types from `.proto` files](xref:grpc/basics#generated-c-assets) using a bundled native binary called `protoc`. Additional steps are required to build gRPC apps on platforms that aren't supported by `Grpc.Tools` bundled native binaries, such as Alpine Linux.
```
Comment (eerhardt) [inline aspnetcore/grpc/troubleshoot.md:289]:
```suggestion
The `Grpc.Tools` package [generates .NET types from `.proto` files](xref:grpc/basics#generated-c-assets) using a bundled native binary called `protoc`. Additional steps are required to build gRPC apps on platforms that aren't supported by the native binaries in `Grpc.Tools`, such as Alpine Linux.
```

**Review by eerhardt — APPROVED** [review]
Thanks!

For my learning, why is the content duplicated?

File: aspnetcore/grpc/troubleshoot/includes/troubleshoot3-7.md [hunk 3]
```diff
@@ -277,6 +277,50 @@ var reply = await client.SayHelloAsync(new HelloRequest { Name = ".NET" });
 
 Alternatively, a client factory can be configured with `Http3Handler` by using <xref:Microsoft.Extensions.DependencyInjection.HttpClientBuilderExtensions.AddHttpMessageHandler%2A>.
 
+## Building gRPC on Alpine Linux
+
+The `Grpc.Tools` package [generates .NET types from `.proto` files](xref:grpc/basics#generated-c-assets) using a bundled native binary called `protoc`. Additional steps are required to build gRPC apps on platforms that aren't supported by `Grpc.Tools` bundled native binaries, such as Alpine Linux.
```
Comment (JamesNK) [inline aspnetcore/grpc/troubleshoot/includes/troubleshoot3-7.md:282]:
```suggestion
The `Grpc.Tools` package [generates .NET types from `.proto` files](xref:grpc/basics#generated-c-assets) using a bundled native binary called `protoc`. Additional steps are required to build gRPC apps on platforms that aren't supported by the native binaries in `Grpc.Tools`, such as Alpine Linux.
```

File: aspnetcore/grpc/troubleshoot/includes/troubleshoot3-7.md [hunk 4]
```diff
@@ -277,6 +277,50 @@ var reply = await client.SayHelloAsync(new HelloRequest { Name = ".NET" });
...
+# e.g. for Alpine Linux the grpc-plugins package can be used
+#  See https://pkgs.alpinelinux.org/package/edge/community/x86_64/grpc-plugins
+apk add grpc-plugins  # Alpine Linux specific package installer
+
+# Set environment variables for the built/installed protoc
+# and grpc_csharp_plugin binaries
+export PROTOBUF_PROTOC=/usr/bin/protoc
+export GRPC_PROTOC_PLUGIN=/usr/bin/grpc_csharp_plugin
+
+# When the dotnet build runs the Grpc.Tools NuGet package will
```
Comment (JamesNK) [inline aspnetcore/grpc/troubleshoot/includes/troubleshoot3-7.md:317]:
```suggestion
# When dotnet build runs, the Grpc.Tools NuGet package will
```

**Comment by JamesNK** [comment]
> Thanks!
> 
> For my learning, why is the content duplicated?

docs folks like to have separate copies of some pages for different .NET versions.

File: aspnetcore/grpc/troubleshoot.md [hunk 5]
```diff
@@ -284,6 +284,50 @@ var reply = await client.SayHelloAsync(new HelloRequest { Name = ".NET" });
...
+#  See https://pkgs.alpinelinux.org/package/edge/community/x86_64/grpc-plugins
+apk add grpc-plugins  # Alpine Linux specific package installer
+
+# Set environment variables for the built/installed protoc
+# and grpc_csharp_plugin binaries
+export PROTOBUF_PROTOC=/usr/bin/protoc
+export GRPC_PROTOC_PLUGIN=/usr/bin/grpc_csharp_plugin
+
+# When dotnet build runs, the Grpc.Tools NuGet package will
+# use the binaries pointed to by the environment variables
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/troubleshoot.md:325]:
```suggestion
# When dotnet build runs, the Grpc.Tools NuGet package
# uses the binaries pointed to by the environment variables.
```

**Review by Rick-Anderson — APPROVED** [review]
Just suggestions, not sure they're improvements so feel free to ignore. It's just the way I'd write it.

**Pull request merged**